### PR TITLE
first approach

### PR DESCRIPTION
**Pull request by VeroCodingProjects**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -9,9 +9,11 @@
     <body>
         <div id="searchBox" class="text-center">
             <h1>Dictionary</h1><br>
-            <input class="form-control" placeholder="Add your word" id="userInput" onfocus=this.value='' autocomplete="on"/>
-            <button id = "addButton" class="btn btn-danger" onclick="addWords()">Add word</button><br>
-            <input class="form-control" placeholder="Search your word" onkeyup="showResults(value)">
-            <div id="result"></div>
+            <form action="javascript:addWords(this)">
+                <input id="input"class="form-control" placeholder="Add your word" onfocus=this.value='' autocomplete="on"/>
```
Comment (keepcoding-tech) [inline index.html]: Here `onfocus=this.value=''` I would change to reset the value after the word was added

File: index.html [hunk 2]
```diff
@@ -9,9 +9,11 @@
...
+                <button class="btn btn-danger">Add word</button><br>
+                <input class="form-control" placeholder="Search your word" onkeyup="showResults(value)">
+                <div id="result"></div>
```
Comment (keepcoding-tech) [inline index.html:18]: You missed to add the closing tag of the div (`</div`)

File: main.js [hunk 3]
```diff
@@ -1,25 +1,27 @@
-const words = []; //the list will contain the added words
+const words = []; 
 
-function addWords() {
-    let userInput = document.getElementById("userInput").value;
-    words.push(userInput);
+function addWords(formData) {
+    if (words.includes(formData.input.value)) {
```
Comment (keepcoding-tech) [inline main.js:5]: This does not seem right, we can still add more words of the same input.

image

File: main.js [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
 }
   
-function showResults(val) { // printing the result
+function showResults(val) {
     let res = document.getElementById("result");
     res.innerHTML = '';
     let list = '';
     let terms = searchInput(val);
-    if (terms.length == 0) { // if the word doesn't exist, then it should be added to the dictionary
+    if (terms.length == 0) {
```
Comment (keepcoding-tech) [inline main.js]: You should return here if there is no word available.

File: main.js [hunk 5]
```diff
@@ -1,25 +1,27 @@
...
         }
     });
 }
   
-function showResults(val) { // printing the result
+function showResults(val) {
     let res = document.getElementById("result");
     res.innerHTML = '';
     let list = '';
     let terms = searchInput(val);
```
Comment (keepcoding-tech) [inline main.js:29]: Let's keep only this variable (`terms`), try to get rid of the rest.

File: style.css [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    background-image: url("https://elements-video-cover-images-0.imgix.net/files/3a6e02dc-2769-456d-a9d7-248ffe6d4005/inline_image_preview.jpg?auto=compress%2Cformat&fit=min&h=394&w=700&s=38a9dc8d5b2e88fda977f4c701b2f620");
+}
+
+input {
+    margin-top: 1cm;
+}
+
+ul {
+    list-style-type: none;
+}
```
Comment (keepcoding-tech) [inline style.css:33]: Try to put the "Add word" button on the same line with the text filed for the input

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

**Comment by keepcoding-tech** [comment]
One last thing, try to append the word to the list after it was added and display it on the page.

File: main.js (hunk 3 above)
Comment (VeroCodingProjects) [reply main.js:5]: This is the last version. The improved one does not have this problem. I think you've watched the wrong branch. :D

File: main.js [hunk 7]
```diff
@@ -1,29 +1,32 @@
-const words = []; //the list will contain the added words
+const words = [];
 
-function addWords() {
-    let userInput = document.getElementById("userInput").value;
-    words.push(userInput);
+function addWords(formData) { //adds the words into the list and clears the input
```
Comment (keepcoding-tech) [inline main.js]: Write the comments above the function

File: main.js [hunk 8]
```diff
@@ -1,29 +1,32 @@
...
+    if (words.includes(formData.input.value)) {
```
Comment (keepcoding-tech) [inline main.js:5]: Let's also return a message that says "the word already exists" or something like that

File: main.js [hunk 9]
```diff
@@ -1,29 +1,32 @@
...
+        return 0;
+    } else {
```
Comment (keepcoding-tech) [inline main.js]: Do you really need this "else"?

File: main.js [hunk 10]
```diff
@@ -1,29 +1,32 @@
...
+    formData.input.value = '';
 }
 
-function searchInput(input) { // searching for all the words that start with the input
-    let reg = input;
+function searchInput(input) { // searching for the subsequence in the list
     return words.filter(function(term) {
-        if (term.startsWith(reg)) {
+        if (term.startsWith(input)) {
             return term;
```
Comment (keepcoding-tech) [inline main.js:20]: Let's return all the words that starts with the input, not just the first one

File: main.js [hunk 11]
```diff
@@ -1,29 +1,32 @@
...
-function showResults(val) { // printing the result
-    let res = document.getElementById("result");
-    res.innerHTML = '';
-    let list = '';
+function showResults(val) { // displays the matches
     let terms = searchInput(val);
-    if (terms.length == 0) { // if the word doesn't exist, then it should be added to the dictionary
-        list += "This word does not exist. Please, add it to the dictionary! :)"
+    if (val == '') {
+        return 0;
```
Comment (keepcoding-tech) [inline main.js:35]: Let's also empty the list here

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: index.html [hunk 12]
```diff
@@ -9,9 +9,14 @@
...
-            <div id="result"></div>
+            <form action="javascript:addWords(this)">
+                <input id="input" placeholder="Add your word" autocomplete="on"/>
+                <button id="button" class="btn btn-dark">Add word</button><br>
+                <h4 id="repetitive"></h4>
+                <h3>Search a word in the dictionary</h3>
+                <input placeholder="Search your word" onkeyup="showResults(value)">
+                <div id="result"></div>
+                <h5 id="dictionary">WORDS IN DICTIONARY: </h5>
+            </form>
```
Comment (keepcoding-tech) [inline index.html:20]: You missed the closing tag here: `</div>`

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

**Pull request merged**